### PR TITLE
Update haskell lsp and hie bios

### PR DESCRIPTION
**Pull request by fendor**
Supersedes both #67 and #74, is based on #67, checks some checkboxes in #48.
ping @alanz, @jneira
Circle ci reports: https://circleci.com/gh/fendor/workflows/haskell-ide-engine/tree/update-haskell-lsp-and-hie-bios
Current status: 3 failing test-cases

### CODE REVIEW

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs [hunk 1]
```diff
@@ -215,9 +215,14 @@ loadCradle iniDynFlags (NewCradle fp) def action = do
...
             -- be that slow, even though the cradle isn't cached because the
             -- `.hi` files will be saved.
-          Right () -> do
+          Right BIOS.Succeeded -> do
             setCurrentCradle cradle
-            logm $ "Cradle set succesfully"
+            logm "Cradle set succesfully"
+            IdeResultOk <$> action
+
+          Right BIOS.Failed -> do
```
Comment (fendor) [inline hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:210]: @mpickering Not sure how to handle this correctly, especially in combination with https://github.com/mpickering/haskell-ide-engine/pull/79/files#diff-fb69f899fd005326a13dddab5748049fR199

**Review by alanz — APPROVED** [review]
LGTM

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs (hunk 1 above)
Comment (jneira) [reply hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:210]: Mmm, i would say that cradle did load succesfully in case of SourceError

Comment (fendor) [reply hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:210]: dunno what reason might exist that BIOS.Failed is returned. I dont know `loadTargets'` enough.

**Comment by jneira** [comment]
Mmm 1572d49 caused timeouts running tests for all ghc's

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs [hunk 2]
```diff
@@ -215,9 +215,14 @@ loadCradle iniDynFlags (NewCradle fp) def action = do
...
+            setCurrentCradle cradle
+            logm "Cradle did not load succeesfully"
```
Comment (jneira) [inline hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs]: typo succeesfully -> succesfully

Comment (lukel97) [inline hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs]:
```suggestion
            logm "Cradle did not load successfully"
```

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs (hunk 1 above)
Comment (lukel97) [reply hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:210]: Should this just return a `IdeResultFail`?

File: hie-plugin-api/hie-plugin-api.cabal [hunk 3]
```diff
@@ -49,10 +49,10 @@ library
...
                      , ghc
-                     , hie-bios
+                     , hie-bios == 0.3.*
                      , ghc-project-types >= 5.9.0.0
                      , cabal-helper
-                     , haskell-lsp == 0.18.*
+                     , haskell-lsp == 0.19.*
                      , hslogger
                      , unliftio
                      , monad-control
```
Comment (lukel97) [inline hie-plugin-api/hie-plugin-api.cabal:58]: Can you also remove `rope-utf16-splay >= 0.3.1.0` from this too? I don't think its needed anymore

**Review by lukel97 — APPROVED** [review]

File: hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs (hunk 1 above)
Comment (fendor) [reply hie-plugin-api/Haskell/Ide/Engine/ModuleCache.hs:210]: No, it is probably fine as it is. It means, that the we could not load the whole module graph, but we can still cache every module up to the point some module failed to typecheck.

**Comment by fendor** [comment]
In my opinion, ready for merge. Two test-cases are failing, one due to caching issues and we are not sure what the other test-case is actually supposed to do at the moment.
https://app.circleci.com/jobs/github/fendor/haskell-ide-engine/2175/tests

**Review by jneira — APPROVED** [review]

**Review by lukel97 — APPROVED** [review]

**Pull request merged**